legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/runtime/SourceCoordinates.java

```diff
@@ -100,4 +127,71 @@ public int getEndColumn()
     {
         return this.endColumn;
     }
+
+    public Preview getPreview()
+    {
+        return this.preview;
+    }
+
+    public static class Preview
+    {
+        private final String beforeText;
+        private final String afterText;
+        private final String foundText;
+
+        public Preview(String beforeText, String foundText, String afterText)
+        {
+            this.beforeText = beforeText;
+            this.afterText = afterText;
+            this.foundText = foundText;
+        }
+
+        @JsonCreator
+        public static Preview newPreview(
+                @JsonProperty("beforeText") String beforeText,
+                @JsonProperty("foundText") String foundText,
+                @JsonProperty("afterText") String afterText
+        )
+        {
+            return new Preview(beforeText, foundText, afterText);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            int hash = this.beforeText.hashCode();
+            hash = 31 * hash + this.foundText.hashCode();
+            hash = 31 * hash + this.afterText.hashCode();
+            return hash;
```

Comment:
Since these fields can be null, you need to use Objects.hashCode.
```suggestion
            int hash = Objects.hashCode(this.beforeText);
            hash = 31 * hash + Objects.hashCode(this.foundText);
            hash = 31 * hash + Objects.hashCode(this.afterText);
            return hash;
```

Or you could just do this:
```suggestion
            return Objects.hash(this.beforeText, this.foundText, this.afterText);
```